tests/python/contrib/test_ethosu/test_encode_constants.py

```diff
@@ -270,5 +273,47 @@ def _get_func():
     assert reference_const_sizes == test_const_sizes
 
 
+def test_constant_as_input():
+    """Test to check that constants specified as inputs aren't
+    interpreted as an encoded constant."""
+
+    def get_graph():
+        dtype = "uint8"
```

Comment:
Why does the constant need to be`uint8`? (just asking for enlightenment)

Reply:
For now this needs to be `uint8` due to https://github.com/apache/tvm/blob/main/python/tvm/relay/backend/contrib/ethosu/tir_to_cs_translator.py#L254. Removing this opens up another can of worms that I'll address in another PR :)

Reply:
Yes, makes sense, thanks for clarifying! :)